Source/Core/Common/Arm64Emitter.cpp

```diff
@@ -784,6 +571,18 @@ void ARM64XEmitter::EncodeLogicalImmInst(u32 op, ARM64Reg Rd, ARM64Reg Rn, u32 i
           (DecodeReg(Rn) << 5) | DecodeReg(Rd));
 }
 
+void ARM64XEmitter::EncodeLogicalImmInst(u32 op, ARM64Reg Rd, ARM64Reg Rn, LogicalImm imm)
+{
+  ASSERT_MSG(DYNAREC, imm.valid, "Invalid logical immediate");
+
+  // Sometimes Rd is fixed to SP, but can still be 32bit or 64bit.
+  // Use Rn to determine bitness here.
+  bool b64Bit = Is64Bit(Rn);
+
+  Write32((b64Bit << 31) | (op << 29) | (0x24 << 23) | (imm.n << 22) | (imm.r << 16) |
+          (imm.s << 10) | (DecodeReg(Rn) << 5) | DecodeReg(Rd));
```

Comment:
Please just call the other implementation of `EncodeLogicalImmInst` instead. Having this constants twice makes it mor error prone.